qrenderdoc/Windows/Dialogs/VirtualFileDialog.cpp

```diff
@@ -134,7 +134,7 @@ class RemoteFileModel : public QAbstractItemModel
       if(normPath[0] != QLatin1Char('/'))
       {
         qCritical() << "Malformed/unexpected path" << path;
-        return QModelIndex();
+        return homeFolder();
```

Comment:
This is not the best place to return `homeFolder` as a default. For example - when the user types in a location to go to, the code calls `indexForPath` to get the index for it, and if it returns an invalid index then an error is shown to the user. With this change the user would be mysteriously redirected to their home folder on any typo.

Instead we only want to default to `homeFolder` if the first initial default directory provided is invalid. So I was suggesting was that the code in the `VirtualFileDialog` construct would be changed like so:

```
  QModelIndex index;

  if(!initialDirectory.isEmpty())
    index = m_Model->indexForPath(initialDirectory);

  if(!index.isValid())
    index = m_Model->homeFolder();

  // switch to home folder and expand it
  changeCurrentDir(index);
```

Then first it would try to fetch the initial directory, if that failed (or if there was no initial directory) it would instead use the home folder.